packages/sfui/shared/scrollable/scrollable.ts

```diff
@@ -133,8 +141,53 @@ export default class Scrollable {
     }
   }
 
-  private onMouseUp() {
+  private onMouseUp(event: MouseEvent) {
+    const { container, options } = this;
     this.isDragged = false;
+
+    // sensitivy drags by how much you drag from x1 -> x1, then swipe inside container by this ammound
+    const sensitivity =
+      typeof this.options.drag === 'object' && this.options.drag.sensitivity ? this.options.drag.sensitivity : 4;
+    // container width options define that no matter how hard you drag x1 -> x2 you will swipe by one container width
+    const scrollByOneWidth = typeof this.options.drag === 'object' ? this.options.drag.containerWidth : false;
+    // buffor for nor overshoot whole width/height of container becuase of subpixel, snap will cover unsufficient width/height
+    const buffor = 10;
+
+    if (options.direction === SfScrollableDirection.vertical) {
+      const element = event.pageY - container.offsetTop;
+      const scrolling = (element - this.dragScrollY) * sensitivity;
+
+      if (scrollByOneWidth) {
+        // if user just touched not dragged at least 10px - do nothing
+        if (Math.abs(this.dragDistance) < 10) return;
+        container.scrollLeft =
+          this.dragScrollLeft -
+          (this.dragDistance < 0 ? container.clientHeight - buffor : -container.clientHeight + buffor);
+      } else {
+        container.scrollTop = this.dragScrollTop - scrolling;
+      }
+    } else {
+      const element = event.pageX - container.offsetLeft;
+      const scrolling = (element - this.dragScrollX) * sensitivity;
+
+      if (scrollByOneWidth) {
+        // if user just touched not dragged at least 10px - do nothing
+        if (Math.abs(this.dragDistance) < 10) return;
+        container.scrollLeft =
+          this.dragScrollLeft -
+          (this.dragDistance < 0 ? container.clientWidth - buffor : -container.clientWidth + buffor);
+      } else {
+        container.scrollLeft = this.dragScrollLeft - scrolling;
+      }
+    }
+
+    // TODO: add proper scrolling effect with timing and easings, now we have no idea when animation ends, therefor cant disable double click during animation and weird behavior. With proper scrolling animation we will be able to add real time dragging not only dragging after mouse/touch UP
```

Comment:
because of that we have hard time predict if we are dragging/scrolling or not. Imagine situation when user drag and again drag fast, now its bug and we can't do much about it, because we dont know if previous drag finished or its in middle of it.